solc/CommandLineInterface.cpp

```diff
@@ -760,8 +763,68 @@ bool CommandLineInterface::processInput()
 			m_compiler->useMetadataLiteralSources(true);
 		if (m_args.count(g_argInputFile))
 			m_compiler->setRemappings(m_args[g_argInputFile].as<vector<string>>());
-		for (auto const& sourceCode: m_sourceCodes)
-			m_compiler->addSource(sourceCode.first, sourceCode.second);
+
+		if (m_args.count(g_argImportAst))
+		{
+			//read file contents, which either are
+			// a) a json-file with multiple sources
+			// b) a json-file with one source only
+			Json::Reader reader;
+			map<string, Json::Value const*> sourceJsons; // will be given to the compilerimportfunction
+			map<string, string> tmp_sources; //used to generate the onchainmetadata-hash
+			map<string, string> supplementarySourceCodes;
+			for (auto const& srcPair: m_sourceCodes) // aka <string, string>
+			{
+				Json::Value* ast = new Json::Value(); //use shared-Pointer here?
+				// try parsing as json
+				if (reader.parse(srcPair.second, *ast, false))
+				{
+					//case a)
+					if ((*ast).isMember("sourceList") && (*ast).isMember("sources"))
+					{
+						for (auto& src: (*ast)["sourceList"])
+						{
+							solAssert(
+								strcmp(
+									(*ast)["sources"][src.asCString()]["AST"]["nodeType"].asCString(),
+									"SourceUnit"
+								) == 0,
+								"the top-level node of the AST needs to be a 'SourceUnit'"
+							);
+							sourceJsons[src.asCString()] = &((*ast)["sources"][src.asString()]["AST"]);
```

Comment:
there is no meaningful answer to this....maybe because of the entropy of the universe is increasing